nixos/modules/system/boot/loader/systemd-boot/systemd-boot-builder.py

```diff
@@ -258,10 +259,13 @@ def main() -> None:
         if available_match is None:
             raise Exception("could not determine systemd-boot version")
 
-        installed_version = installed_match.group(1)
-        available_version = available_match.group(1)
+        installed_version = version.parse(installed_match.group(1))
+        available_version = version.parse(available_match.group(1))
 
-        if installed_version < available_version:
+        # systemd 252 has a regression that leaves some machines unbootable, so we skip that update.
+        # The fix is in 252.2
+        # See https://github.com/systemd/systemd/issues/25363 and https://github.com/NixOS/nixpkgs/pull/201558#issuecomment-1348603263
+        if installed_version < available_version and available_version > version.parse('252.1'):
```

Comment:
It's also fine if systemd is a 251.

Setting `systemd.package` can be used to influence with version is available here, so we need to explicitly check for the bad versions. And probably log if we skip because we know its bad.